File: answers/lintcode/Length of Last Word.cpp
class Solution {
public:
    /**
     * @param s A string
     * @return the length of last word
     */
    int lengthOfLastWord(string& s) {
        // Write your code here
		if (s.length() == 0) {
			return 0;
		}
		bool start = false;
		size_t left = 0;
		size_t right = s.length() - 1;
		for (size_t i = s.length() - 1; i < s.length(); -- i) {
			if (' ' != s[i] && ! start) {
				right = i;
				start = true;
			}
			else if (' ' == s[i] && start) {
				left = i + 1;
				break;
			}
		}
		return right - left + 1;
    }
};

File: answers/leetcode/Length of Last Word/Length of Last Word.py
class Solution(object):
    def lengthOfLastWord(self, s):
        """
        :type s: str
        :rtype: int
        """

        begin = 0
        end = 0
        for i in reversed(range(len(s) + 1)):
            if i < len(s) and ' ' != s[i] and (0 == i or ' ' == s[i - 1]):
                begin = i
                break
            elif (len(s) == i or ' ' == s[i]) and i > 0 and ' ' != s[i - 1]:
                end = i
        return end - begin
